tests/console/zfs.pm

```diff
@@ -0,0 +1,190 @@
+    scrub('tank');
+    assert_script_run("zpool status tank | grep scan | grep 'scrub repaired'");
+    assert_script_run('md5sum -c /var/tmp/Big_Buck_Bunny_8_seconds_bird_clip.ogv.md5sum');
+    script_run('zpool status tank');
+    assert_script_run('zpool status tank | grep "corrupted data"');
+    # Replace disk
+    assert_script_run('zpool offline tank /var/tmp/tank_a.img');
+    assert_script_run('zpool replace tank /var/tmp/tank_a.img');
+    scrub('tank');
+    assert_script_run('zpool status tank | grep state | grep ONLINE');
+    # Display status for debugging purposes
+    script_run('zpool status tank');
+    save_screenshot();
+    ## Test Snapshots
+    assert_script_run('zfs snapshot tank@initial');
+    assert_script_run('zfs list -t snapshot | grep "tank@initial"');
+    assert_script_run('curl -v -o /tank/test_unzip.zip ' . data_url('console/test_unzip.zip'));
+    assert_script_run('rm /tank/Big_Buck_Bunny_8_seconds_bird_clip.ogv');
+    # Create second snapshot with other data
+    assert_script_run('zfs snapshot tank@second');
+    assert_script_run('zfs snapshot tank@third');
+    assert_script_run('zfs list -t snapshot | grep "tank@second"');
+    assert_script_run('zfs list -t snapshot | grep "tank@third"');
+    # Mount snapshots
+    assert_script_run('mkdir -p /mnt/tank/{initial,second}');
+    assert_script_run('mount -t zfs tank@initial /mnt/tank/initial');
+    assert_script_run('mount -t zfs tank@second /mnt/tank/second');
+    assert_script_run('stat /mnt/tank/initial/Big_Buck_Bunny_8_seconds_bird_clip.ogv');
+    assert_script_run('! stat /mnt/tank/initial/test_unzip.zip');
+    assert_script_run('! stat /mnt/tank/second/Big_Buck_Bunny_8_seconds_bird_clip.ogv');
+    assert_script_run('stat /mnt/tank/second/test_unzip.zip');
+    assert_script_run('umount /mnt/tank/{initial,second}');
+    # Test holding snapshots and delete snapshots
+    assert_script_run('zfs hold keep tank@initial');
+    assert_script_run('zfs hold keep tank@second');
+    assert_script_run('! zfs destroy tank@initial');
+    assert_script_run('! zfs destroy tank@second');
+    assert_script_run('zfs release -r keep tank@second');
+    assert_script_run('zfs destroy tank@second');
+    script_retry('zfs list -t snapshot | grep -v "tank@second"', delay => 1, retry => 30);
+    assert_script_run('zfs list -t snapshot | grep "tank@initial"');
+    assert_script_run('zfs list -t snapshot | grep "tank@third"');
+    # Rename snapshot
+    assert_script_run('zfs rename tank@initial tank@today');
+    assert_script_run('zfs list -t snapshot | grep "tank@today"');
+    assert_script_run('! zfs list -t snapshot | grep "tank@initial"');
+    # Send snapshot to second zfs pool
+    assert_script_run('zfs send "tank@today" | zfs recv -F "dozer@today"');
+    script_run('zfs list -t snapshot | grep "dozer@today"');
+    ## Test rollback
+    assert_script_run('zfs rollback -r tank@today');
+    # Rollback to today also deletes more recent snaptshots
```

Comment:
I don't see any rollback here.

Reply:
That comment belongs is a note to the applied rollback: `zfs rollback -r tank@today`

I updated the comment to make that better understandable